src/Suave/Http.fs

```diff
@@ -786,3 +786,81 @@ module Http =
             }
       }
       |> succeed
+
+  module CORS =
+    
+    open System
+    open Successful
+
+    [<Literal>]
+    let origin = "Origin"
+    [<Literal>]
+    let accessControlRequestMethod = "Access-Control-Request-Method"
+    [<Literal>]
+    let accessControlRequestHeaders = "Access-Control-Request-Headers"
+    [<Literal>]
+    let accessControlAllowOrigin = "Access-Control-Allow-Origin"
+    [<Literal>]
+    let accessControlAllowMethods = "Access-Control-Allow-Methods"
+    [<Literal>]
+    let accessControlAllowHeaders = "Access-Control-Allow-Headers"
+    [<Literal>]
+    let accessControlAllowCredentials = "Access-Control-Allow-Credentials"
+    [<Literal>]
+    let accessControlExposeHeaders = "Access-Control-Expose-Headers"
+    [<Literal>]
+    let accessControlMaxAge = "Access-Control-Max-Age"
+
+    let cors (allowedUris : string list) : WebPart =
+        fun (ctx : HttpContext) ->
+            let req = ctx.request
+            match req.header origin with
```

Comment:
Still not sure how to handle Choice.bind. I don't understand how to use that correctly.